masonry/src/widget/text_area.rs

```diff
@@ -734,8 +788,17 @@ impl<const EDITABLE: bool> Widget for TextArea<EDITABLE> {
 
     fn update(&mut self, ctx: &mut UpdateCtx, event: &Update) {
         match event {
-            Update::FocusChanged(_) => {
-                ctx.request_render();
+            Update::FocusChanged(focused) => {
+                // HACK: currently, when moving focus away from a text area, the Ime::Disabled
+                // event is routed to the newly focused widget. Do IME clean up here.
+                if !focused && self.editor.is_composing() {
+                    let (fctx, lctx) = ctx.text_contexts();
+                    self.editor.transact(fctx, lctx, |txn| txn.clear_compose());
+                    self.rendered_generation = self.editor.generation();
+                    ctx.request_layout();
+                } else {
+                    ctx.request_render();
+                }
             }
```

Comment:
This is a bit unfortunate.

Reply:
I think sending `Ime::Disabled` to the newly unfocused widget in the focus update pass would be a reasonable solution here. It does mean that the new widget gets `Disabled, Enabled`. But that's probably fine?

Reply:
I still think this would be the right approach. It does make our pass code slightly more complicated, but makes the IME handling much more consistent.

Reply:
Yes, I think this is better. The focus update pass now sends a synthesized `Ime::Disabled` event (if IME was active) just prior to switching focus to the next focused widget.